catalog/lib/widgets/app_base_button.dart

```diff
@@ -0,0 +1,89 @@
+import 'package:catalog/catalog.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+
+class AppBaseButton extends StatelessWidget {
+  final String text;
+  final VoidCallback? onPressed;
+  final Color backgroundColor;
+  final Color textColor;
+  final Color focusColor;
+  final Color disabledTextColor;
+  final Color pressedColor;
+  final Color disabledColor;
+  final Color hoveredColor;
+  final Color? borderSideColor;
+  final Icon? iconLeft;
+  final Icon? iconRight;
+  final double verticalPadding;
+
+  const AppBaseButton({
+    required this.text,
+    required this.onPressed,
+    required this.backgroundColor,
+    required this.textColor,
+    required this.disabledColor,
+    required this.focusColor,
+    required this.pressedColor,
+    required this.disabledTextColor,
+    required this.hoveredColor,
+    this.borderSideColor,
+    this.iconLeft,
+    this.iconRight,
+    this.verticalPadding = 12,
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context) => MaterialButton(
```

Comment:
Material button will be deprecated in the future https://api.flutter.dev/flutter/material/MaterialButton-class.html
We need to change this to use the new Material3 [components](https://m3.material.io/components/all-buttons)

Reply:
Another alternative is creating them manually with other widgets, like Container, but that might be more work than to use the new material buttons

Reply:
Shouldn't we try to use material buttons as they come and set the theme with the default style for each one? instead of having a base class?

Reply:
Something like this in AppTheme:
```
 return ThemeData(
      fontFamily: FontFamily.youngSerif,
      primaryColor: _colors.primary,
      filledButtonTheme: _styles.getFilledButtonThemeData(),
      ...
      
```